lib/binding_rust/lib.rs

```diff
@@ -2353,27 +2353,26 @@ impl Query {
                 ffi::TSQueryErrorNodeType | ffi::TSQueryErrorField | ffi::TSQueryErrorCapture => {
                     let suffix = source.split_at(offset).1;
                     let in_quotes = source.as_bytes()[offset - 1] == b'"';
-                    let mut end_offset = suffix.len();
-                    if let Some(pos) = suffix
-                        .char_indices()
-                        .take_while(|(_, c)| *c != '\n')
-                        .find_map(|(i, c)| match c {
-                            '"' if in_quotes
-                                && i > 0
-                                && suffix.chars().nth(i - 1) != Some('\\') =>
-                            {
-                                Some(i)
-                            }
-                            c if !in_quotes
-                                && (c.is_whitespace() || c == '(' || c == ')' || c == ':') =>
-                            {
-                                Some(i)
+                    let end_offset = suffix
+                        .find({
+                            let mut backslashes = 0;
+                            move |c| {
+                                if in_quotes {
```

Comment:
you can move the assignment of backslashes outside of the closure, and the closure doesn't have to explicitly `move` anything, otherwise this looks great